tests/ui/collapsible_str_replace.rs

```diff
@@ -0,0 +1,68 @@
+// run-rustfix
+
+#![warn(clippy::collapsible_str_replace)]
+
+fn get_filter() -> &'static str {
+    "u"
+}
+
+fn main() {
+    let misspelled = "hesuo worpd";
+
+    let p = 'p';
+    let s = 's';
+    let u = 'u';
+    let l = "l";
+
+    // LINT CASES
+    let _ = misspelled.replace('s', "l").replace('u', "l");
+
+    let _ = misspelled.replace('s', l).replace('u', l);
+
+    let _ = misspelled.replace('s', "l").replace('u', "l").replace('p', "l");
+
+    let _ = misspelled
+        .replace('s', "l")
+        .replace('u', "l")
+        .replace('p', "l")
+        .replace('d', "l");
+
+    let _ = misspelled.replace(s, "l").replace('u', "l");
+
+    let _ = misspelled.replace(s, "l").replace('u', "l").replace('p', "l");
+
+    let _ = misspelled.replace(s, "l").replace(u, "l").replace('p', "l");
+
+    let _ = misspelled.replace(s, "l").replace(u, "l").replace(p, "l");
+
+    let _ = misspelled.replace('s', "l").replace('u', "l").replace('p', "x");
+
+    let _ = misspelled.replace('s', "x").replace('u', "l").replace('p', "l");
+
+    // NO LINT CASES
+    let _ = misspelled.replace('s', "l");
+
+    let _ = misspelled.replace(s, "l");
```

Comment:
Every test case with only one `replace` is more or less redundant. This is because the code won't even get to the point that it checks the `replace` arguments. I think the test you wrote are viable, but you should add another `replace` to the single-replace cases.